### ISSUE TITLE
libdb2.dylib

### ISSUE THREAD

**Issue by alaindewilde** [ibmdb/python-ibmdb#333]
Despite the export command as described many help messages and the original readme , i'm not able to solve the error i get in notebook at "import ibm_db" command  

ImportError: dlopen(/anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg/ibm_db.cpython-36m-darwin.so, 2): Library not loaded: libdb2.dylib
  Referenced from: /anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg/ibm_db.cpython-36m-darwin.so
  Reason: image not found

I'm using Mac , installad anaconda , started jupyter notebook . Executed the EXPORT command referring the libdb2.dylib library in a command windows , but no success. 

What is the correct order , where should i execute what , how can i check the correct EXPORT command was issued, ...

**Comment by SabaKauser** [comment]
you can see if this helps.
https://www.ibm.com/developerworks/community/blogs/96960515-2ea1-4391-8170-b0515d08e4da/entry/ibm_db_on_MAC_OS_Hints_and_Tips?lang=en

**Comment by alaindewilde** [comment]
Thanks. Tried that one , no result. Document does not describe what to execute where (command line, notebook) and how i can verify correct export command is given.

**Comment by SabaKauser** [comment]
I had run it from MAC Command line. 
as far as I remember, install_name_tool -change was needed for MAC OS to correctly locate libdb2.dylib.
on Linux, ldd could show the dependencies, and setting LD_LIBRARY_PATH would make the OS to pick correct path for loading dynamic lib.
However, MAC seem to have an extra security setting that prevents it from loading the libs from relative path or DYLD_LIBRARY_PATH setting would not have effect and having to use install_name_tool would make the dependencies accessible as if they are present in the current path.
There was another user who had posted this -> https://github.com/ibmdb/python-ibmdb/pull/221/files

**Comment by alaindewilde** [comment]
I guess i don't have enough knowledge on all this, just wanted to connect from a jupyter notebook to Db2 but i'm not getting anywhere with only commands. Thanks for your help, but I give up.

**Comment by SzickA** [comment]
@alaindewilde 
1, Open the terminal
2, cd /anaconda3/lib/python3.6/site-packages/
3, install_name_tool -change libdb2.dylib `pwd`/libdb2.dylib ibm_db.cpython-36m-darwin.so
4, ln -s /Applications/dsdriver/lib/libdb2.dylib libdb2.dylib

**Comment by alaindewilde** [comment]
Install_name_tool command did not give an error after changing line 1 to 
1) cd /anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg/

But that did not change the initial behavior in the notebook : 
Requirement already satisfied: ibm_db in /anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg (2.0.8)

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-3ab5bfbae856> in <module>()
     15 #
     16 
---> 17 import ibm_db
     18 import pandas
     19 import ibm_db_dbi

ImportError: dlopen(/anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg/ibm_db.cpython-36m-darwin.so, 2): Library not loaded: pwd/libdb2.dylib
  Referenced from: /anaconda3/lib/python3.6/site-packages/ibm_db-2.0.8-py3.6-macosx-10.7-x86_64.egg/ibm_db.cpython-36m-darwin.so
  Reason: image not found

Thanks all for attempts to help, but don't bother. I stop my attempts. To many issues to get a notebook working.

**Comment by SzickA** [comment]
@alaindewilde 
You shouldn't change the path, install name tool should ran from the /anaconda3/lib/python3.6/site-packages folder

Andras

**Comment by lechie** [comment]
> you can see if this helps.
> https://www.ibm.com/developerworks/community/blogs/96960515-2ea1-4391-8170-b0515d08e4da/entry/ibm_db_on_MAC_OS_Hints_and_Tips?lang=en

it works fine for me , thank you

**Comment by nearodney** [comment]
@SabaKauser it is pure fiction that these workarounds work for Mac OS X High Sierra 10.13.6 and you should not close all these open issues. You should resolve them.

**Comment by Geobuddy** [comment]
https://medium.com/@sudhanvalp/overcome-ibm-db-import-error-image-not-found-on-macos-578f07b70762 - Great solution for mac.

**Comment by oscarsan** [comment]
I get that working but aftet that i just get this error, python 3.8.7. Osx Catalina

    Importing test library '/Users/C943186/dev/op/xapi/xapi-mt/xapi-acceptance-test/robotframework/libs/Xct.py' failed: 
    ImportError: dlopen(/Users/C943186/.pyenv/versions/3.8.7/lib/python3.8/site-packages/ibm_db.cpython-38-darwin.so, 2): 
    Symbol not found: ___cxa_throw_bad_array_new_length
       Referenced from: /Users/C943186/.pyenv/versions/3.8.7/lib/python3.8/site-packages/clidriver/lib/libdb2.dylib
       Expected in: /usr/lib/libstdc++.6.dylib
    in /Users/C943186/.pyenv/versions/3.8.7/lib/python3.8/site-packages/clidriver/lib/libdb2.dylib

**Comment by bimalkjha** [comment]
@oscarsan Please uninstall ibm_db and after that delete `/Users/C943186/.pyenv/versions/3.8.7/lib/python3.8/site-packages/clidriver` directory if it exist. After that reinstall ibm_db```. It should download new clidriver and issue should get fixed.
If you still see issue post reinstall, please share the full result of below commands:
```
cd /Users/C943186/.pyenv/versions/3.8.7/lib/python3.8/site-packages/clidriver/lib
db2level
echo $DYLD_LIBRARY_PATH
otool -L libdb2.dylib
```
Thanks.

**Comment by oscarsan** [comment]
@bimalkjha  actually this issue was fixed for me in the new 3.0.3 package
https://github.com/ibmdb/python-ibmdb/issues/608#issuecomment-799508100
Just needed to reinstall it.

**Comment by bimalkjha** [comment]
@oscarsan Good to know issue is fixed. I commented after pushing updated clidriver for ibm_db which has the fix. Thanks.
